golem-component-service-base/src/api/dto.rs

```diff
@@ -0,0 +1,210 @@
+use crate::model::plugin as local_plugin_model;
+use crate::model::plugin::PluginWasmFileReference;
+use golem_common::model::plugin as common_plugin_model;
+use golem_common::model::plugin::{PluginOwner, PluginScope};
+use golem_service_base::poem::TempFileUpload;
+use poem_openapi::Multipart;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, poem_openapi::Union)]
```

Comment:
These are moved here and separated for two reasons:

* we now have different types for creation between the rest and grpc api. There is now a domain type that unifies both that they convert to in the api layer.
* We have a different representation for plugins in the rest and grpc api as I don't want to expose blob storage paths through the rest api -- they have no meaning for external users.